tests/lib.rs

```diff
@@ -387,3 +387,30 @@ fn foreach() {
         _ => unreachable!(),
     });
 }
+
+#[test]
+fn retain() {
+    // do same operations on a plain vector
+    // to verify retain implementation
+    let mut v = Vec::new();
+    let (r, mut w) = evmap::new();
+
+    for i in 0..50 {
+        w.insert(0, i);
+        v.push(i);
+    }
+
+    w.retain(0, |num| num % 2 == 0).refresh();
+    v.retain(|num| num % 2 == 0);
```

Comment:
I think I'd factor out the closure here since the whole point is that they should be exactly the same.